content/developers/developer-patterns/3rdparty-verification/index.md

```diff
@@ -0,0 +1,366 @@
+    veracity \
+    replicate-logs --replicadir merklelogs
+  ```
+
+  {{< /tab >}}
+  {{< /tabs >}}
+
+{{< note>}} Take care with larger time horizons, it may trigger issues with rate limiting.{{< /note >}}
+
+If a copy of the event data recorded by DataTrails is retained, or available at the time of dispute, then the tenants event data is Tamper Proof.
+
+The remainder of this article discusses what the commands `replicate-logs` and `watch` in more depth, covers how to replicate only selective tenants, and explains the significance of the replicated materials.
+
+## How Veracity Supports These Guarantees
+
+See [Independently verifying DataTrails transparent merkle logs](/developers/developer-patterns/veracity/) for a general introduction to `veracity`.
+
+This diagram illustrates the logical flow when updating a local replica using veracity.
+
+---
+
+```mermaid
+---
+config:
+    theme: classic
+---
+sequenceDiagram
+    actor v as Verifier
+    box Runs locally to the verifier
+    participant V as Veracity
+    participant R as Replica
+    end
+    participant D as DataTrails
+
+    v -->> V: Safely update my replica to massif X please
+    V ->> D: Fetch and verify the remote massifs and seals up to X
+    V ->> R: Check the verified remote data is consistent with the replica
+    V ->> R: Update the replica with verified additions
+    V -->> v: All OK!
+```
+
+---
+
+The guarantees of *non-falsifiability* and *non-repudiation* require replication and verification of at least the most recently updated massif.
+They further require that the replica is updated often enough to capture all massifs.
+As a massif, in the default tenant configuration, contains over 16,000 events, the frequency necessary to support this guarantee is both low, and completely determined by the specific tenant of interest.
+
+The guarantees of *provability* and *demonstrable completeness* require retention of any local massif that contains an event that is still of value.
+
+Saving the API response data when events are recoded, or obtaining the event data using the DataTrails events API, is additionally required in order to support a fully audit for data corruption.
+
+With a trusted local copy of the verifiable data, even after a tamper is detected, it is reasonable to rely on DataTrails storage of the event data.
+
+When the event is fetched, if it can be verified against the replica, it proves that the DataTrails storage of the event remains correct.
+If it does not, it is proven that both the DataTrails log and a tenants merkle log have been improperly updated.
+
+DataTrails is a platform for sharing *meta* data. And typically, a DataTrails event contains a commitment (a hash of) some other material: An AI Model Card, An SBOM, or an image or video. DataTrails does not handle the original material at all.
+
+In the extreme case, where the DataTrails database, and or ledger, is corrupted, relying parties can instantly detect this.
```

Comment:
I think this comment has lost its context. I have, perhaps mistakenly, taken it as read that a reader of this document understands there is a regular saas store *and* a ledger. That seems like such a fundamental piece of the picture maybe we need a diffferent section for it ? And one we can refer to in all the places this will come up